forge-game/src/main/java/forge/game/ability/effects/UntapAllEffect.java

```diff
@@ -30,12 +38,19 @@ public void resolve(SpellAbility sa) {
         }
         list = CardLists.getValidCards(list, valid, sa.getActivatingPlayer(), card, sa);
 
-        boolean remember = sa.hasParam("RememberUntapped");
+        CardCollection untapped = new CardCollection();
         for (Card c : list) {
-            c.untap(true);
-            if (remember) {
-                card.addRemembered(c);
+            if (c.untap(true))  {
+                untapped.add(c);
+                if (sa.hasParam("RememberUntapped")) card.addRemembered(c);
             }
         }
+        if (!untapped.isEmpty()) {
+            final Map<AbilityKey, Object> runParams = AbilityKey.newMap();
+            final Map<Player, CardCollection> map = Maps.newHashMap();
+            map.put(activator, untapped);
```

Comment:
though it doesn't matter that much yet there can be a different untapper (I suspect only a few thankfully)
e. g. _Early Harvest_

Reply:
Think I've got this sorted

Reply:
Your refactor would break cards like _Awakening_
it really depends on the card

Reply:
_Twist Allegiance_ should also be a single untap